homeassistant/components/axis/binary_sensor.py

```diff
@@ -23,26 +26,159 @@
+    """Axis binary sensor entity description."""
+
+    event_topic: tuple[EventTopic, ...] | EventTopic
+    """Event topic that provides state updates."""
+    name_fn: Callable[[AxisHub, Event], str] = lambda hub, event: ""
+    """Function providing the corresponding name to the event ID."""
+    supported_fn: Callable[[AxisHub, Event], bool] = lambda hub, event: True
+    """Function validating if event is supported."""
+
+
+@callback
+def event_id_is_int(event_id: str) -> bool:
+    """Make sure event ID is int."""
+    try:
+        return isinstance(int(event_id), int)
+    except ValueError:
+        return False
+
+
+@callback
+def guard_suite_supported_fn(hub: AxisHub, event: Event) -> bool:
+    """Validate event ID is int."""
+    _, _, profile_id = event.id.partition("Profile")
+    return event_id_is_int(profile_id)
+
+
+@callback
+def object_analytics_supported_fn(hub: AxisHub, event: Event) -> bool:
+    """Validate event ID is int."""
+    _, _, profile_id = event.id.partition("Scenario")
+    return event_id_is_int(profile_id)
+
+
+@callback
+def guard_suite_name_fn(
+    handler: FenceGuardHandler
+    | LoiteringGuardHandler
+    | MotionGuardHandler
+    | Vmd4Handler,
+    event: Event,
+    event_type: str,
+) -> str:
+    """Get guard suite item name."""
+    if handler.initialized and (profiles := handler["0"].profiles):
+        for profile_id, profile in profiles.items():
+            camera_id = profile.camera
+            if event.id == f"Camera{camera_id}Profile{profile_id}":
+                return f"{event_type} {profile.name}"
+    return ""
+
+
+@callback
+def fence_guard_name_fn(hub: AxisHub, event: Event) -> str:
+    """Fence guard name."""
+    return guard_suite_name_fn(hub.api.vapix.fence_guard, event, "Fence Guard")
+
+
+@callback
+def loitering_guard_name_fn(hub: AxisHub, event: Event) -> str:
+    """Loitering guard name."""
+    return guard_suite_name_fn(hub.api.vapix.loitering_guard, event, "Loitering Guard")
+
+
+@callback
+def motion_guard_name_fn(hub: AxisHub, event: Event) -> str:
+    """Motion guard name."""
+    return guard_suite_name_fn(hub.api.vapix.motion_guard, event, "Motion Guard")
+
+
+@callback
+def motion_detection_4_name_fn(hub: AxisHub, event: Event) -> str:
+    """Motion detection 4 name."""
+    return guard_suite_name_fn(hub.api.vapix.vmd4, event, "VMD4")
+
+
+@callback
+def object_analytics_name_fn(hub: AxisHub, event: Event) -> str:
+    """Get object analytics name."""
+    if hub.api.vapix.object_analytics.initialized and (
+        scenarios := hub.api.vapix.object_analytics["0"].scenarios
+    ):
+        for scenario_id, scenario in scenarios.items():
+            device_id = scenario.devices[0]["id"]
+            if event.id == f"Device{device_id}Scenario{scenario_id}":
+                return f"Object Analytics {scenario.name}"
+    return ""
+
+
+ENTITY_DESCRIPTIONS = (
+    AxisBinarySensorDescription(
+        key="Input port state",
```

Comment:
Shouldn't these be actual keys? Like `input_port_state`?

Reply:
You tell me :), I don't use them for anything so I just did it the same way as I did with all of descriptions of the UniFi integration